### PR TITLE
Added GENERATED AS IDENTITY support to DataBricks dialect

### LINKED ISSUE #4455: Databricks - Columns with Identity
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`L:   1 | P:  41 |  PRS | Line 1, Position 41: Found unparsable section`

For a table creation with a column using
`GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1)`

https://docs.databricks.com/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html

for example:

CREATE TABLE IF NOT EXISTS person (
    person_id bigint GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1)
)

### Expected Behaviour

should be a parsable section.

### Observed Behaviour

fails to parse table creation with IDENTITY (START WITH 1 INCREMENT BY 1)

### How to reproduce

Run sqlfluff on the example table creation in the description.

### Dialect

Databricks

### Version

sqlfluff:1.4.5

### Configuration

sqlfluff lint  --dialect databricks --exclude-rules L029

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by nicolb2305**
### Brief summary of the change made
* Added support for the DataBricks specific GENERATED AS IDENTITY syntax as described [here](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html). Fixes #4455
* Added corresponding test cases to the DataBricks dialect.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by nicolb2305** [comment]
Saw that you were okay with being tagged for a pull request that fixes #4455, @alanmcruickshank

File: src/sqlfluff/dialects/dialect_databricks.py [hunk 1]
```diff
@@ -377,3 +378,58 @@ class AliasExpressionSegment(sparksql.AliasExpressionSegment):
...
+class GeneratedColumnDefinitionSegment(sparksql.GeneratedColumnDefinitionSegment):
+    """A generated column definition, e.g. for CREATE TABLE or ALTER TABLE.
+
+    https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html
+    """
+
+    match_grammar: Matchable = Sequence(
+        Ref("SingleIdentifierGrammar"),  # Column name
+        Ref("DatatypeSegment"),  # Column type
+        Bracketed(Anything(), optional=True),  # For types like VARCHAR(100)
```
Comment (hashemi-soroush) [inline src/sqlfluff/dialects/dialect_databricks.py:392]:
```suggestion
        Bracketed(Anything(), optional=True),  # For types like DECIMAL(3, 2)
```
Databricks has no `VARCHAR` type.

Comment (nicolb2305) [reply src/sqlfluff/dialects/dialect_databricks.py:392]: I believe DataBricks does let you define columns as varchar, but just converts it to a string under the hood.

Comment (nicolb2305) [reply src/sqlfluff/dialects/dialect_databricks.py]: Changed it as suggested anyway, as I do agree that using VARCHAR in Databricks is quite unusual.

File: src/sqlfluff/dialects/dialect_databricks.py [hunk 2]
```diff
@@ -377,3 +378,58 @@ class AliasExpressionSegment(sparksql.AliasExpressionSegment):
...
+                "GENERATED",
+                OneOf(
+                    "ALWAYS",
+                    Sequence("BY", "DEFAULT"),
+                ),
+                "AS",
+                "IDENTITY",
+                Bracketed(
+                    Sequence(
+                        Sequence(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_databricks.py:415]: is it possible these two can be the other way round?

Comment (nicolb2305) [reply src/sqlfluff/dialects/dialect_databricks.py:415]: I didn't think about that, but I just checked, and it seems "STARTS WITH" does have to come before "INCREMENT BY".

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**